### PR TITLE
Unpin channels-redis

### LINKED ISSUE #332: Update to v4 results in "RuntimeError: Event loop is closed"
After upgrading to `channels-redis==4.0.0`, our celery tasks are all reporting the following traceback:
```
future: <Task finished name='Task-9' coro=<Connection.disconnect() done, defined at /usr/local/lib/python3.9/site-packages/red                                                                                                                                                                is/asyncio/connection.py:819> exception=RuntimeError('Event loop is closed')>
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/redis/asyncio/connection.py", line 828, in disconnect
    self._writer.close()  # type: ignore[union-attr]
  File "/usr/local/lib/python3.9/asyncio/streams.py", line 353, in close
    return self._transport.close()
  File "/usr/local/lib/python3.9/asyncio/selector_events.py", line 698, in close
    self._loop.call_soon(self._call_connection_lost, None)
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 751, in call_soon
    self._check_closed()
  File "/usr/local/lib/python3.9/asyncio/base_events.py", line 515, in _check_closed
    raise RuntimeError('Event loop is closed')
RuntimeError: Event loop is closed
[2022-10-12 07:30:31,972] [ERROR] [asyncio] Task exception was never retrieved
```

Downgrading the image to `channels-redis==3.4.1` resolves the issue, so I'm starting out here.  This seems probably related to #312.  

Image OS is Debian Bullseye, amd64.  The django application is running with gunicorn.

Probably related packages:
```
celery==5.2.7
channels==3.0.5
channels-redis==4.0.0
hiredis==2.0.0
redis==4.3.4
```
Full Pipfile.lock: https://github.com/paperless-ngx/paperless-ngx/blob/dev/Pipfile.lock

### PR DESCRIPTION
**Pull request by jessicamack**
##### SUMMARY

The bug that initially caused the upgrade block has been resolved https://github.com/django/channels_redis/issues/332.
[As a result aioredis was removed because version 4.0.0 of channel-redis migrated the underlying Redis library from aioredis to redis-py.](https://github.com/django/channels_redis/blob/13cef4502a95fbb7f41bb97eaa6f4d02c1f7c91c/CHANGELOG.txt#L37).

##### ISSUE TYPE

 - Bug, Docs Fix or other nominal change

##### COMPONENT NAME

 - Other

##### AWX VERSION

```
awx: 24.6.2
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: requirements/requirements.in [hunk 1]
```diff
@@ -7,7 +7,7 @@ azure-keyvault
 boto3
 botocore
 channels
-channels-redis==3.4.1  # see UPGRADE BLOCKERs
+channels-redis
```
Comment (thedoubl3j) [inline requirements/requirements.in:10]: Do we want to fully unpin this given its upgrade blocker status before?

Comment (TheRealHaoLiu) [reply requirements/requirements.in:10]: its fine

File: awx/main/wsrelay.py [hunk 2]
```diff
@@ -191,7 +191,7 @@ async def run_producer(self, name, websocket, group):
                         return
 
                     continue
-                except aioredis.errors.ConnectionClosedError:
+                except redis.exceptions.ConnectionError:
```
Comment (TheRealHaoLiu) [inline awx/main/wsrelay.py:202]: this might not be the right 1 to 1 replacement of the exception

Comment (jessicamack) [reply awx/main/wsrelay.py:202]: This was the exception I saw used in the aioredis code base where the ConnectionClosedError was before but I'm open to other options

Comment (TheRealHaoLiu) [reply awx/main/wsrelay.py:202]: this should be ok

**Review by thedoubl3j — APPROVED** [review]

File: requirements/README.md [hunk 3]
```diff
@@ -120,15 +120,6 @@ in the top-level Makefile.
...
-### channels-redis
-
-Due to an upstream bug (linked below), we see `RuntimeError: Event loop is closed` errors with newer versions of `channels-redis`.
-Upstream is aware of the bug and it is likely to be fixed in the next release according to the issue linked below.
-For now, we pin to the old version, 3.4.1
-
-* https://github.com/django/channels_redis/issues/332
-* https://github.com/ansible/awx/issues/13313
-
 ### hiredis
```
Comment (TheRealHaoLiu) [inline requirements/README.md]: This blocker is also removed in this PR

**Comment by Sharvesh0306** [comment]
@TheRealHaoLiu when will 24.6.2 be released?

**Pull request merged**